libs/zopengl/src/wrapper.zig

```diff
@@ -759,7 +978,17 @@ pub fn getError() Error {
 }
 
 // pub var getFloatv: *const fn (pname: Enum, data: [*c]Float) callconv(.C) void = undefined;
+
 // pub var getIntegerv: *const fn (pname: Enum, data: [*c]Int) callconv(.C) void = undefined;
+pub fn getInteger(parameter: Parameter) Int {
+    var res: Int = undefined;
+    bindings.getIntegerv(@enumToInt(parameter), &res);
+    return res;
+}
+pub fn getIntegerv(parameter: Parameter, ptr: [*]Int) void {
+    bindings.getIntegerv(@enumToInt(parameter), ptr);
+}
```

Comment:
This is awful. Is it really impossible to get some safety here? Is there some way to find out how many bytes opengl is going to from the pointer?

Reply:
I wanted to see what WebGL does here. And well, it doesn't have these functions. Instead it has a `getParameter` that returns "typed" thing(s) depending on the pname.

Maybe we should try something similar using comptime generics?

Reply:
I've revised this in https://github.com/michal-z/zig-gamedev/pull/242/commits/ebec59163b96dfc2f3140bcc4f08b313b2de48bb
I think this works well, at least on user's side
````zig
// pub var getIntegerv: *const fn (pname: Enum, data: [*c]Int) callconv(.C) void = undefined;
pub fn getInteger1(pname: ParameterName1) Int {
    var res: Int = undefined;
    bindings.getIntegerv(@enumToInt(pname), &res);
    return res;
}
pub fn getInteger2(pname: ParameterName2) [2]Int {
    var res: [2]Int = undefined;
    bindings.getIntegerv(@enumToInt(pname), &res);
    return res;
}
pub fn getInteger4(pname: ParameterName4) [4]Int {
    var res: [4]Int = undefined;
    bindings.getIntegerv(@enumToInt(pname), &res);
    return res;
}
```
What do you think?

Reply:
This doesn't work. Some parameters return a variable amount of things. `GL_COMPRESSED_TEXTURE_FORMATS` for example.